packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -260,91 +285,208 @@ function transformTerraformOutput(terraformOutput: string): DeployAirnodeOutput
+  const { type, region } = cloudProvider as CloudProvider;
+
+  const spinner = logger.getSpinner().start(`Deploying Airnode ${airnodeAddress} ${stage} to ${type} ${region}`);
   if (logger.inDebugMode()) {
     spinner.info();
   }
 
-  const { type: cloudProviderType } = cloudProvider;
-  const bucket = getBucketName(airnodeAddressShort, stage);
-  const terraformStateCloudProviderDir = path.join(terraformStateDir, cloudProviderType);
-
-  if (!(await cloudProviderLib[cloudProviderType].stateExists(bucket, cloudProvider as any))) {
-    // Run state recipes
-    logger.debug('Running state Terraform recipes');
-    const stateTmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'airnode'));
-
-    const execOptions = { cwd: stateTmpDir };
-    await execTerraform(execOptions, 'init', [['from-module', terraformStateCloudProviderDir]]);
-
-    const commonArguments: CommandArg[] = [
-      ['var', 'airnode_address_short', airnodeAddressShort],
-      ['var', 'stage', stage],
-      ['input', 'false'],
-      'no-color',
-      'auto-approve',
-    ];
-    await execTerraform(execOptions, 'apply', prepareAirnodeManageArguments(cloudProvider, commonArguments));
+  const goDeploy = await go(async () => {
+    logger.debug('Fetching Airnode bucket');
+    let bucketName = await cloudProviderLib[type].getAirnodeBucket(cloudProvider as any);
+    if (!bucketName) {
+      logger.debug('No Airnode bucket found, creating');
+      bucketName = await cloudProviderLib[type].createAirnodeBucket(cloudProvider as any);
+    }
+    logger.debug(`Using Airnode bucket '${bucketName}'`);
+
+    logger.debug('Fetching Airnode bucket content');
+    const directoryStructure = await cloudProviderLib[type].getBucketDirectoryStructure(
+      cloudProvider as any,
+      bucketName
+    );
+
+    const timestamp = Date.now();
+    const bucketStagePath = `${airnodeAddress}/${stage}`;
+    const bucketDeploymentPath = `${bucketStagePath}/${timestamp}`;
+
+    const stageDirectory = getStageDirectory(directoryStructure, airnodeAddress, stage);
+    if (stageDirectory) {
+      logger.debug(`Deployment '${bucketStagePath}' already exists`);
+      const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
+      const bucketConfigPath = `${bucketStagePath}/${latestDeployment}/config.json`;
+      logger.debug(`Fetching configuration file '${bucketConfigPath}'`);
+      const remoteConfig = JSON.parse(
+        await cloudProviderLib[type].getFileFromBucket(cloudProvider as any, bucketName, bucketConfigPath)
+      ) as Config;
+
+      const remoteNodeSettings = remoteConfig.nodeSettings;
+      const remoteCloudProvider = remoteNodeSettings.cloudProvider as CloudProvider;
+      if (remoteNodeSettings.nodeVersion !== nodeVersion) {
+        throw new Error(
+          `Can't update an Airnode deployment with airnode-deployer of a different version. Deployed version: ${remoteNodeSettings.nodeVersion}, airnode-deployer version: ${nodeVersion}`
```

Comment:
Depends on what you consider updating. It won't affect changing a configuration of a deployed Airnode (redeployment). But you won't be able to update from let's say 0.8.0 to 0.9.0. That should be done by removing the old Airnode (with 0.8.0 deployer) and deploying the new one (with 0.9.0 deployer).

Reply:
Oh, I see. And when you remove the Airnode it will remove the directory structure, so this check will not be reached and Airnode will be deployed sucessfully.

Reply:
The directory structure is common for all the Airnodes of all versions. You can have for example one Airnode with address A of version 0.8.0 and one with address B of version 0.9.0. The check checks only the specific Airnode (it's `config.json` to be precise) against the version of the deployer running the deployment/removal.

Reply:
I mean it will remove the directory structure for the specific Airnode address (not the whole bucket folder)

Reply:
Right, yes, that it will.